sunsynk/sensor.py

```diff
@@ -241,17 +241,90 @@ def update_value(self) -> None:
             _LOGGER.error("Could not decode temperature: %s", err)
 
 
+class Minutes(int):
+    """Deals with inverter time format conversion complexities."""
+
+    @classmethod
+    def from_str_value(cls, str_value: str) -> Minutes:
+        """Create from a string in hh:mm format."""
+        return cls(int(str_value[:-3]) * 60 + int(str_value[-2:]))
+
+    @classmethod
+    def from_reg_value(cls, reg_value: int) -> Minutes:
+        """Create from a register value."""
+        hours, minutes = divmod(reg_value, 100)
+        return cls(hours * 60 + minutes)
+
+    @property
+    def reg_value(self) -> int:
+        """Get the register value."""
+        hours, minutes = divmod(self, 60)
+        return hours * 100 + minutes
+
+    @property
+    def str_value(self) -> str:
+        """Get the value in hh:mm format."""
+        hours, minutes = divmod(self, 60)
+        return f"{hours}:{minutes:02}"
+
+
+@attr.define(slots=True)
 class TimeRWSensor(RWSensor):
     """Extract the time."""
 
+    min: TimeRWSensor = attr.field(default=None)
+    max: TimeRWSensor = attr.field(default=None)
+
+    @property
+    def minutes(self) -> Minutes:
+        """Get the value of this sensor as total minutes."""
+        return Minutes.from_reg_value(self.reg_value[0])
+
+    def available_values(self, step_minutes: int) -> List[str]:
+        """Get the available values for this sensor."""
+        full_day = 24 * 60
+
+        min_val = self.min.minutes if self.min else 0
+        max_val = self.max.minutes if self.max else full_day
+        val = self.minutes
+
+        time_range = self._range(min_val, max_val, val, step_minutes, full_day)
+
+        return list(map(lambda i: Minutes(i).str_value, time_range))
+
+    def dependencies(self) -> List[Sensor]:
+        """Get a list of sensors upon which this sensor depends."""
+        sensors: List[Sensor] = []
+        if isinstance(self.min, TimeRWSensor):
+            sensors.append(self.min)
+        if isinstance(self.max, TimeRWSensor):
+            sensors.append(self.max)
+        return sensors
+
+    def min_max(self, min_value: TimeRWSensor, max_value: TimeRWSensor) -> None:
+        """Set the min and max values."""
+        self.min = min_value
+        self.max = max_value
```

Comment:
taking the other suggestion into account, we probably don't need this. It is also only used once